assets/aml-benchmark/components/dataset-preprocessor/spec.yaml

```diff
@@ -0,0 +1,56 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+name:  dataset_preprocessor 
+display_name: Dataset Preprocessor
+description: Dataset Preprocessor
+version: 0.0.1
+is_deterministic: true
+
+inputs:
+  dataset: 
+    type: uri_file
+    description: >-
+      Path to load the dataset.
+    optional: False
+  template_input:
+    type: string
+    description: >-
+      A json dictionary where key is the name of the column enclosed in " " and associated dict value is presented using jinja template logic 
+      which will be used to extract respective value from the dataset. 
+      Example format:
+      {"<user_column_name>": {{key in the json file for this column}}, ....}. The processed output will be dumped to a 
+      jsonl file in this format: {"<user_column_name>": "<key in the json file for this column>", ....} 
+    optional: True
+  script_path:
+    type: uri_file
+    description: >-
+      Path to the custom preprocessor python script provided by user. If this is provided, then, this takes precedence.  
+    optional: True
+  encoder_config:
+    type: string
+    description: >-
+      Dictionary in json format that contains column/key name to encode associated with dict key `column_name` followed by  
+      idtolabel or labeltoid mappers as key-value pair.
```

Comment:
Would be good to have an example here, too

Reply:
Thank you. Added this as example : {"column_name":"label", "0":"NEUTRAL", "1":"ENTAILMENT", "2":"CONTRADICTION"}.